Applied a few changes in the credential section

Max-Meng opened the pull request:
This document described two types of credentials: server-level credential and database scoped credential.  There are a few issues in the current version of the document: 
1. It omits 'server-level' somewhere it shouldn't, which caused confusion. 
2. It only provides instructions about how to grant permission to create and use server-level credential, but not for database scoped credential. 
3. It calls 'server-level credential' somewhere, but calls it 'server-scoped credential' somewhere else, which caused confusion.

Max-Meng committed:
Applied a few changes in the credential section

This document described two types of credentials: server-level credential and database scoped credential. 
There are a few issues in the current version of the document: 
1. It omits 'server-level' somewhere it shouldn't, which caused confusion. 
2. It only provides instructions about how to grant permission to create and use server-level credential, but not for database scoped credential. 
3. It calls 'server-level credential' somewhere, but calls it 'server-scoped credential' somewhere else, which caused confusion.

JasonWHowell commented on articles/synapse-analytics/sql/develop-storage-files-storage-access-control.md:
````diff
@@ -236,25 +236,39 @@ To query a file located in Azure Storage, your serverless SQL pool end point nee
+```sql
+GRANT REFERENCES ON DATABASE SCOPED CREDENTIAL::[database-scoped_credential] TO [specific_user];
+```
+
````
Please add a manual anchor to fix all the broken links, right above the heading. 

```suggestion
<a name="server-scoped-credential"></a>
```

JasonWHowell replied:
@ShannonLeavitt  can you accept this suggestion to fix the broken anchor links?

ShannonLeavitt replied:
@JasonWHowell - Looks like this is already resolved. But our team can commit suggestions if needed for future PRs. If you need us to do that, you or the author can add the request to the sign-off comment.

Max-Meng committed:
Update overview-features.md

1. Update term server-scoped-credential to server-level-credential to make it consistent with other documents.  
2. Fix a bookmark reference issue caused by the term change. (https://github.com/MicrosoftDocs/azure-docs/pull/107906)

Max-Meng committed:
Update query-delta-lake-format.md

Fix a bookmark reference issue caused by the term change. (MicrosoftDocs#107906)

Max-Meng committed:
Update query-json-files.md

Max-Meng committed:
Update query-parquet-files.md

Fix a bookmark reference issue caused by the term change. (https://github.com/MicrosoftDocs/azure-docs/pull/107906)

Max-Meng committed:
Update query-single-csv-file.md

Fix a bookmark reference issue caused by the term change. (https://github.com/MicrosoftDocs/azure-docs/pull/107906)

Max-Meng committed:
Update resources-self-help-sql-on-demand.md

Fix a bookmark reference issue caused by the term change. (https://github.com/MicrosoftDocs/azure-docs/pull/107906)
